README.md

```diff
@@ -282,6 +339,44 @@ flask 0.5     PYSEC-2019-179 1.0          Successfully upgraded flask (0.5 => 1.
 flask 0.5     PYSEC-2018-66  0.12.3       Successfully upgraded flask (0.5 => 1.0)
 ```
 
+## Troubleshooting
+
+Have you resolved a problem with `pip-audit`? Help us by contributing to this
+section!
+
+### `pip-audit` takes longer than I expect!
+
+Depending on how you're using it, `pip-audit` may have to perform its
+own dependency resolution, which can take roughly as long as `pip install`
+does for a project. See the [security model](#security-model) for an explanation.
+
+You have two options for avoiding dependency resolution: *audit a pre-installed
+environment*, or *ensure that your dependencies are already fully resolved*.
+
+If you know that you've already fully configured an environment equivalent
+to the one that `pip-audit -r requirements.txt` would audit, you can simply
+reuse it:
+
+```console
+# Note the absence of any "input" arguments, indicating that the environment is used.
+$ pip-audit
+
+# Optionally filter out non-local packages, for virtual environments:
+$ pip-audit --local
+```
+
+Alternatively, if your input is fully pinned (and optionally hashed), you
+can tell `pip-audit` to skip dependency resolution with either `--no-deps`
+(pinned without hashes) or `--require-hashes` (pinned including hashes):
```

Comment:
Might be good to link to https://pip.pypa.io/en/stable/cli/pip_install/#hash-checking-mode here.